### ISSUE TITLE
no physical Sources - SoundTouch 520

### ISSUE THREAD

**Issue by Bingo2023** [thlucas1/homeassistantcomponent_soundtouchplus#11]
### System Health details

no information here

### Checklist

- [X] I have enabled SmartInspect debug logging for my installation.
- [X] I have filled out the issue template to the best of my ability.
- [X] This issue only contains 1 issue (if you have multiple issues, open one issue for each issue).
- [X] This issue is not a duplicate issue of any [previous issues](https://github.com/thlucas1/homeassistantcomponent_soundtouchplus/issues?q=is%3Aissue+label%3A%22Bug%22+)..

### Describe the issue

hello Lucas,
thanks for the good work you do here. I really appreciate it, as the original Bose integration kind of sucks :slight_smile: 

I have an (see also pic attached):
BOSE SoundTouch 520 5.1 Home Cinema System

Unfortunately it the source selection is not working as expected, it does not show any of the physical inputs. As you see, the remote has 6 input buttons (5x HDMI, 1x BT&Digital media).

I feel like if I switch via old remote control to e.g. GAME the HA integrations (official Bose integration & Bose SoundTouchPlus) consider the system to be OFF. I guess that the Amplifier turn off the black bar (marked red in pic).
 Never the less, SoundTouch app and Win-Version are working excellent.

See below what I get as source displayed with latest version v1.0.26.
Attaches are also the logs from browser.
![2024-01-02_19h43_00](https://github.com/thlucas1/homeassistantcomponent_soundtouchplus/assets/61052575/d007e8ee-d6a1-4f96-b012-56b29ffb587d)

![th-2845783390](https://github.com/thlucas1/homeassistantcomponent_soundtouchplus/assets/61052575/4964b517-d2e6-4961-8a10-e256cae31d22)
![s-l500-3451630720](https://github.com/thlucas1/homeassistantcomponent_soundtouchplus/assets/61052575/d88ee7ed-c290-4de4-8a58-a56899e9f1f9)

[sources_info_suppURLs.zip](https://github.com/thlucas1/homeassistantcomponent_soundtouchplus/files/13813221/sources_info_suppURLs.zip)

### Reproduction steps

1.
2.
3.
...

### Debug logs

```text
see zip file
```

**Comment by thlucas1** [comment]
Hi Bingo - The `AUX` and `BLUETOOTH` source are not showing up in the sources.xml that you listed.  I am not sure why that is.  Do you have any devices connected to those ports (AUX, HDMI 1-5) on the 520?  I would try selecting a source (e.g. AUX or HDMI1) while something is playing thru the device, and then reissue the command to retrieve the sources.xml.  The 520 may be temporarily adding and removing the source as it needs it.  I say "may be", as I have never tested with a 520 device before so I am not sure.  As for BLUETOOTH, try playing a BlueTooth source from your phone or other BT device, and then reissue the command to retrieve the sources.xml.

Also, even though a source does not appear in the CONFIG list of sources, it should still show in the media player as the actual source value that is selected.  You can verify the source that is playing by issuing a call to the [/nowPlaying](https://github.com/thlucas1/homeassistantcomponent_soundtouchplus/wiki/SoundTouch-WebServices-API#now-playing-status) service from your home computer browser (e.g. `http://192.168.x.x:8090/nowPlaying`).  Output should be something like this:
```
<?xml version="1.0" encoding="UTF-8" ?>
<nowPlaying deviceID="xxxxxxxx" source="BLUETOOTH" sourceAccount="">
  <ContentItem source="BLUETOOTH" location="" sourceAccount="" isPresetable="false">
    <itemName>My iPhone</itemName>
  </ContentItem>
  <track>Baby, It&apos;s Cold Outside (Duet With Michael Bublé)</track>
  <artist>Various Artists</artist>
  <album>Christmas Mix</album>
  <stationName>My iPhone</stationName>
  <art artImageStatus="SHOW_DEFAULT_IMAGE" />
  <skipEnabled />
  <playStatus>PLAY_STATE</playStatus>
  <skipPreviousEnabled />
  <genre>Christmas</genre>
  <connectionStatusInfo status="CONNECTED" deviceName="My iPhone" />
</nowPlaying>
```

You can also try selecting the source from the developer tools services menu - example yaml (change yourdeviceid to your 520 entity_id):
```
service: media_player.select_source
data:
  source: BLUETOOTH
target:
  entity_id: media_player.YOURDEVICEID
```

Do all of your sources show up in the HA soundtouch (not soundtouchplus) integration?  

Here is how my sources show up in my HA instance using the standard HA media player:
```
type: media-control
entity: media_player.bose_st10_1
```
![image](https://github.com/thlucas1/homeassistantcomponent_soundtouchplus/assets/38439576/de4fbe40-e197-48ca-bcde-9051fb48eab0)

**Comment by Bingo2023** [comment]
> Do all of your sources show up in the HA soundtouch (not soundtouchplus) integration?
no, sources are not working with original soundtouch integration. That is why I investigated to find a better solution.

**Comment by Bingo2023** [comment]
I did this method with the nowPlaying command. See attached the result, it is only showing "local" source.... no matter what HDMI input I take.

Any other idear?



[nowPlaying.zip](https://github.com/thlucas1/homeassistantcomponent_soundtouchplus/files/13818541/nowPlaying.zip)

**Comment by Bingo2023** [comment]
I started to use Wireshark to snif in.... after checking the SoundTouch app deeper, it seems that only the black box is controllable via App, the Amplifier may not - see screenshot below, in the app I can only select the digital sources like NAS and BT.

![2024-01-03_12h59_25](https://github.com/thlucas1/homeassistantcomponent_soundtouchplus/assets/61052575/7e936eb6-5df8-4d4f-8085-7795fe625e24)

**Comment by thlucas1** [comment]
I would agree with you on your statement of "... only the black box (SoundTouch Wireless Adapter) is controllable via the SoundTouch App, the Amplifier may not".  I ran across the following in the Bose 520 manual, that indicates this is probably true:
![image](https://github.com/thlucas1/homeassistantcomponent_soundtouchplus/assets/38439576/a79c6442-2bfe-470b-aecf-2d05fbe6d6b5)

I had another Bose 520 user with the same situation; he was trying to switch the 520 source to play input from an LG TV with webOS.  [Here is the link to that thread](https://community.home-assistant.io/t/sorry-for-the-delay/635159/55) with everything that we tried.  The closest we got was using the following service call to switch to the LOCAL source - it would never switch to the HDMI ARC connection though.  
```
alias: Bose ST-520 Listening Mode - Plex (with TV Sound Toggle)
sequence:
  - service: media_player.select_sound_mode
    alias: select Bose ST-520 Soundbar Source LOCAL
    data:
      sound_mode: LOCAL
    target:
      entity_id: media_player.trash_2
mode: single
icon: mdi:soundbar
```

I thought it was an issue with the TV not switching properly since it was using HDMI ARC processing ... but now that I see what is displayed in the SoundTouch App, it seems that the amp is not controllable as you suggested.  The SoundTouch Wireless Adapter box simply adds SoundTouch sources (e.g. BLUETOOTH, AUX, Pandora, Spotify, TuneIn, etc) that get fed into the amplifier via a dedicated SoundTouch input, and cannot control the other (non SoundTouch) amplifier inputs themselves.

With all of that said, I did discover a `SoundTouchHelper.exe` tool in the SoundTouch App for Windows directory if you want to try it out (located in `C:\Program Files (x86)\SoundTouch\SoundTouchHelper` directory).  Running that will give you the following screen:
![image](https://github.com/thlucas1/homeassistantcomponent_soundtouchplus/assets/38439576/78e5229f-cf9b-4e6b-ba64-7bba6b9e77fd)
Check the SoundTouch device that you want to diagnose, and click on the `Collect New Logs`.  It will generate some logs and package them in a zip file.  Take a look at the `ScmDiagsLogs` folder output for more information.  It's also fun to run a Wireshark trace while that is working, which will give you the same information contained in the logs.

Hope it helps, and I wish I had better news.

**Comment by thlucas1** [comment]
I am updating the project wiki to include [notes for specific SoundTouch device models](https://github.com/thlucas1/homeassistantcomponent_soundtouchplus/wiki/SoundTouch-Model-Notes#soundtouch-520).

Do you have a few minutes to run some tests for me so that I can update the ST-520 section?  I'm looking to see what aspects of the 520 can be controlled via Home Assistant service calls (e.g. volume, power, etc).

Thanks!

### Standard Home Assistant media_player services
```
# power on the device.
service: media_player.turn_on
data: {}
target:
  entity_id: media_player.bose_st10_1

```
```
# power on the device.
service: media_player.turn_off
data: {}
target:
  entity_id: media_player.bose_st10_1

```
```
# turn volume up.
service: media_player.volume_up
data: {}
target:
  entity_id: media_player.bose_st10_1
```
```
# turn volume down.
service: media_player.volume_down
data: {}
target:
  entity_id: media_player.bose_st10_1
```
```
# toggle volume mute
service: media_player.volume_mute
data:
  is_volume_muted: false
target:
  entity_id: media_player.bose_st10_1
```
```
# select LOCAL soundtouch source.
# I don't think this one will work since the 520 does not support /selectLocalSource function.
service: media_player.select_source
data:
  source: LOCAL
target:
  entity_id: media_player.bose_st10_1
```
```
# select BLUETOOTH source.
service: media_player.select_source
data:
  source: BLUETOOTH
target:
  entity_id: media_player.bose_st10_1
```
```
# select AUX source.
service: media_player.select_source
data:
  source: 'AUX:AUX'
target:
  entity_id: media_player.bose_st10_1
```
### soundtouchplus integration services:
```
# remote control power button press and release.
service: soundtouchplus.remote_keypress
data:
  entity_id: media_player.bose_st10_1
  key_id: POWER
  key_state: both
```
```
# remote control preset number 2 select
service: soundtouchplus.remote_keypress
data:
  entity_id: media_player.bose_st10_1
  key_id: PRESET_2
  key_state: release
```

**Comment by Bingo2023** [comment]
sorry needed some time to come over it that my Bose is only semi "Connected".

you wrote that another user w/ 520 did some test.. unfortunately the link is not working.
I would expect if we use the "local" source I could use the SoundTouch Plus integration to turn on/off and it will stay at same HDMI input (would be sufficient for only 1 input device - my case).

I'd like to help you with the documentation. How can I test these commands?

**Comment by thlucas1** [comment]
Here's the link to that thread I was talking about with the other 520 user:
`https://community.home-assistant.io/t/sorry-for-the-delay/635159/57`.

The service commands can be tested from the Home Assistant Developer Tools interface:
![image](https://github.com/thlucas1/homeassistantcomponent_soundtouchplus/assets/38439576/73047f9a-b8e0-4a88-b3da-8427a7f0892a)

Thank you for your help!

**Comment by Bingo2023** [comment]
> I am updating the project wiki to include [notes for specific SoundTouch device models](https://github.com/thlucas1/homeassistantcomponent_soundtouchplus/wiki/SoundTouch-Model-Notes#soundtouch-520).
> 
> Do you have a few minutes to run some tests for me so that I can update the ST-520 section? I'm looking to see what aspects of the 520 can be controlled via Home Assistant service calls (e.g. volume, power, etc).
> 
> Thanks!
> 
> ### Standard Home Assistant media_player services
> ```
> # power on the device.
> service: media_player.turn_on
> data: {}
> target:
>   entity_id: media_player.bose_st10_1
> ```
**works,  520 plays and switches Source from HDMI to "Soundtouch" e.g. internet radio**
> 
> ```
> # power off the device.
> service: media_player.turn_off
> data: {}
> target:
>   entity_id: media_player.bose_st10_1
> ```
**OK**

> 
> ```
> # turn volume up.
> service: media_player.volume_up
> data: {}
> target:
>   entity_id: media_player.bose_st10_1
> ```
**OK**
> 
> ```
> # turn volume down.
> service: media_player.volume_down
> data: {}
> target:
>   entity_id: media_player.bose_st10_1
> ```
**OK**
> 
> ```
> # toggle volume mute.
> service: media_player.volume_mute
> data:
>   is_volume_muted: false
> target:
>   entity_id: media_player.bose_st10_1
> ```
**OK**
> 
> ```
> # select LOCAL soundtouch source.
> # I don't think this one will work since the 520 does not support /selectLocalSource function.
> service: media_player.select_source
> data:
>   source: LOCAL
> target:
>   entity_id: media_player.bose_st10_1
> ```
it does something, internet radio stops playing and Display say "Soundtouch". Or from BT -> "Soundtouch"
> 
> ```
> # select BLUETOOTH source.
> service: media_player.select_source
> data:
>   source: BLUETOOTH
> target:
>   entity_id: media_player.bose_st10_1
> ```
**OK**
> 
> ```
> # select AUX source.
> service: media_player.select_source
> data:
>   source: 'AUX:AUX'
> target:
>   entity_id: media_player.bose_st10_1
> ```
**Fail: Unknown error**

> 
> ### soundtouchplus integration services:
> ```
> # remote control power button press and release.
> service: soundtouchplus.remote_keypress
> data:
>   entity_id: media_player.bose_st10_1
>   key_id: POWER
>   key_state: both
> ```
**turn on/off and plays last inputsource from Soundtouch: BT or internet radio**

> 
> ```
> # remote control preset number 2 select.
> service: soundtouchplus.remote_keypress
> data:
>   entity_id: media_player.bose_st10_1
>   key_id: PRESET_2
>   key_state: release
> ```
**ok w/ PRESET_1 in my case, play music**

**Comment by Bingo2023** [comment]
hello Lucas, below the command you can find the result in **bold**

**Comment by thlucas1** [comment]
Thank you @Bingo2023 - that was very helpful.  I updated my [SoundTouch Model Notes](https://github.com/thlucas1/homeassistantcomponent_soundtouchplus/wiki/SoundTouch-Model-Notes) wiki page with your results, which will hopefully help other 520 owners with future questions.

**Comment by Bingo2023** [comment]
u welcome. Thank you for the good work.

By the way, default name for this device is (could be changed in the wiki):
*.bose_st10_1     ->    *.soundtouch_520_1

**Comment by thlucas1** [comment]
Thank you - I adjusted the 520 section device name yaml example in the [wiki](https://github.com/thlucas1/homeassistantcomponent_soundtouchplus/wiki/SoundTouch-Model-Notes#soundtouch-520).

**Comment by thlucas1** [comment]
@Bingo2023 Thanks for trying those commands, they were worth a shot.  I was pretty sure they were not going to work with the Wireless Adapter (attached to an ST-520), but wanted to confirm.  Those commands will work for the ST-10 and ST-300 though, as I have confirmed, in case anyone follows this thread in the future.

With that said, I think we have tried everything that we can and just have to accept the fact that the Bose Wireless Adapter has limited SoundTouch functionality when it comes to the WebServices API.  I wish Bose had done a better job integrating it with their supporting products, but it at least give you some other sources (Spotify, Pandora, TuneIn, etc) that the original device does not.

Thanks for getting back to me on the results.  I will go ahead and close the GitHub thread for this issue as well.
